src/mlpack/methods/linear_svm/linear_svm_main.cpp

```diff
@@ -0,0 +1,417 @@
+    // Long description.
+    "An implementation of LinearSVM using either the "
+    "L-BFGS optimizer or ParallelSGD (stochastic gradient descent)."
+    "  This solves the classification problem."
+    "\n\n"
+    "This program allows loading a LinearSVM model (via the " +
+    PRINT_PARAM_STRING("input_model") + " parameter) "
+    "or training a LinearSVM model given training data (specified "
+    "with the " + PRINT_PARAM_STRING("training") + " parameter), or both "
+    "those things at once.  In addition, this program allows classification on "
+    "a test dataset (specified with the " + PRINT_PARAM_STRING("test") + " "
+    "parameter) and the classification results may be saved with the " +
+    PRINT_PARAM_STRING("predictions") + " output parameter."
+    " The trained LinearSVM model may be saved using the " +
+    PRINT_PARAM_STRING("output_model") + " output parameter."
+    "\n\n"
+    "The training data, if specified, may have class labels as its last "
+    "dimension.  Alternately, the " + PRINT_PARAM_STRING("labels") + " "
+    "parameter may be used to specify a separate matrix of labels."
+    "\n\n"
+    "When a model is being trained, there are many options.  L2 regularization "
+    "(to prevent overfitting) can be specified with the " +
+    PRINT_PARAM_STRING("lambda") + " option, and The number of classes can be "
+    "manually specified with the " + PRINT_PARAM_STRING("number_of_classes") +
+    "and if an intercept term is not desired in the model, the " +
+    PRINT_PARAM_STRING("no_intercept") + " parameter can be specified."
+    "Margin of difference between correct class and other classes can "
+    "be specified with the" + PRINT_PARAM_STRING("delta") + "option."
+    "The optimizer used to train the model can be specified with the " +
+    PRINT_PARAM_STRING("optimizer") + " parameter.  Available options are "
+    "'psgd' (stochastic gradient descent) and 'lbfgs' (the L-BFGS optimizer).  "
+    "There are also various parameters for the optimizer; the " +
+    PRINT_PARAM_STRING("max_iterations") + " parameter specifies the maximum "
+    "number of allowed iterations, and the " +
+    PRINT_PARAM_STRING("tolerance") + " parameter specifies the tolerance for "
+    "convergence.  For the ParallelSGD optimizer, the " +
+    PRINT_PARAM_STRING("step_size") + " parameter controls the step size taken "
+    "at each iteration by the optimizer. If the "
+    "objective function for your data is oscillating between Inf and 0, the "
+    "step size is probably too large.  There are more parameters for the "
+    "optimizers, but the C++ interface must be used to access these."
+    "\n\n"
+    "We can only use ParallelSGD when OPENMP is specified.  "
+    "For ParallelSGD, an iteration refers to a single point. So to take"
+    " a single pass over the dataset with ParallelSGD, " +
+    PRINT_PARAM_STRING("max_iterations") +
+    " should be set to the number of points in the dataset."
```

Comment:
Most users will expect, when using an SGD-like optimizer, to instead just specify the number of passes over the data.  So, I would suggest that instead of doing one-iteration-per-point here, either we add an `epochs` option which is used for parallel SGD instead of `max_iterations` (meaning `max_iterations` is only used for L-BFGS), or do some processing so that `max_iterations` refers to the number of passes when SGD is used.

Reply:
Added new parameter `epochs`.